common/ferrostar/src/routing_adapters/osrm/algorithms.rs

```diff
@@ -0,0 +1,84 @@
+use super::models::{Annotation, AnnotationValue, MaxSpeed, MaxSpeedUnits};
+use itertools::izip;
+
+/// Get's he slice of annotations
+pub(crate) fn get_annotation_slice(
+    start_index: usize,
+    end_index: usize,
+    annotations: Option<Vec<AnnotationValue>>,
+) -> Option<Vec<AnnotationValue>> {
+    return if let Some(annotations) = &annotations {
+        let annot_len = annotations.len();
```

Comment:
nit: You can use `?` unwrapping to propagate optionals, just like you can to propagate results:

```
let annotations = annotations?;
```

Admittedly I haven't dug into this code much, but I'm also wondering if rather than checking for None in this method, it might be cleaner to move the `None` check up a level, and avoid calling this method altogether in that case.

Reply:
Yeah, it's probably a map pattern ;) I made some comments about that earlier.

Reply:
All of these `Some` ifs and matches are now gone 👍